rasterio/session.py

```diff
@@ -94,6 +94,9 @@ def cls_from_path(path):
         elif path.scheme == "oss" or "aliyuncs.com" in path.path:
             return OSSSession
 
+        elif path.scheme == "swift":
```

Comment:
```suggestion
        elif path.path.startswith("/vsiswift/"):
```

Without a "swift" scheme, this is the proper test.

Reply:
fixed